### DISCUSSION TITLE
How do i read a Channel value from the script?

### DISCUSSION THREAD

**Discussion by vittodevit** [arendst/Tasmota#15010] [Q&A] [answered]
### I am trying to read the RGBW channels value from the script, but it's not working.

I copied the code from the documentation.
![image](https://user-images.githubusercontent.com/58268988/156231349-60ba4dac-0688-4336-95f0-4d750bf3ca91.png)

This is my code.
![image](https://user-images.githubusercontent.com/58268988/156231431-9f2aca63-33d1-4d8d-a6a8-fefc964c3464.png)

In execution it's not working...
![image](https://user-images.githubusercontent.com/58268988/156231703-898de492-0d21-4bfb-baa0-13138616aca1.png)

The binary is self compiled with _TasmoCompiler_ and flashed with _Tasmotizer_. These are the features enabled and the flags.
![image](https://user-images.githubusercontent.com/58268988/156231955-812d993e-f3dc-48b5-888a-dfa8577c5742.png)
![image](https://user-images.githubusercontent.com/58268988/156231975-dffd11de-2e7b-4cb4-b09e-792b199e75c3.png)

If anyone knows anything about this, any help is appreciated ♥

**Comment by sfromis** [comment, marked as answer]
In my shallow understanding, you need to create all variables in the `>D` section before you can use them, as done in the sample code of which a section is visible in your screenshot.

**Reply by vittodevit** [reply]
Yeah i forgot to add the `chan*` variables in the `>D` section, i will try adding them and see if the `Channel[n]` variable works.

**Reply by vittodevit** [reply]
It works! If anyone needs the code for sending the `Color` and `Power` data to a slave Arduino on the serial port, it's this.
```
>D
red=0
green=0
blue=0
white=0

relay1=0
relay2=0
strip=0

>P
relay1=pwr[1]
relay2=pwr[2]
strip=pwr[3]
print SCR: 1->%relay1%, 2->%relay2% S->%strip%

>E
red=Channel[1]
green=Channel[2]
blue=Channel[3]
white=Channel[4]
print SCR: R->%red%, G->%green%, B->%blue%, W->%white%

>F
;RED CHANNEL
if upd[red]>0
then
->SSerialSend 3,%red%
endif

;GREEN CHANNEL
if upd[green]>0
then
->SSerialSend 4,%green%
endif

;BLUE CHANNEL
if upd[blue]>0
then
->SSerialSend 5,%blue%
endif

;WHITE CHANNEL
if upd[white]>0
then
->SSerialSend 6,%white%
endif

;RELAY1 CHANNEL
if upd[relay1]>0
then
->SSerialSend 1,%relay1%
endif

;RELAY2 CHANNEL
if upd[relay2]>0
then
->SSerialSend 2,%relay2%
endif

;STRIP CHANNEL
if upd[strip]>0
then
->SSerialSend 7,%strip%
endif
```
I would also suggest adding in the documentation, in the variables section, the `Channel[x]` variable which is seen only in the examples.

**Reply by sfromis** [reply]
`Channel[x]` is not variables, just a generic parsing of the JSON payloads you already can see in the console. Scripting docs can't really list all what exists in some payload. 
`pwr[x]` is different, as this is a scripting language feature to check power state, and therefore documented specifically, along with other functions.

**Reply by vittodevit** [reply]
> `Channel[x]` is not variables, just a generic parsing of the JSON payloads you already can see in the console. Scripting docs can't really list all what exists in some payload. `pwr[x]` is different, as this is a scripting language feature to check power state, and therefore documented specifically, along with other functions.

Oh i didn't know you could use any payload as a "variable" in the script. That's cool, thanks for the help anyways 😄!
